extra/intl-extra/IntlExtension.php

```diff
@@ -32,6 +32,17 @@ final class IntlExtension extends AbstractExtension
         'medium' => \IntlDateFormatter::MEDIUM,
         'long' => \IntlDateFormatter::LONG,
         'full' => \IntlDateFormatter::FULL,
+        'relative_short' => \IntlDateFormatter::RELATIVE_SHORT,
+        'relative_medium' => \IntlDateFormatter::RELATIVE_MEDIUM,
+        'relative_long' => \IntlDateFormatter::RELATIVE_LONG,
+        'relative_full' => \IntlDateFormatter::RELATIVE_FULL,
```

Comment:
Those must be used conditionally, to avoid breaking things on PHP 7.x

Reply:
Ideally, it would thrown an exception when this formats are used with PHP 7, telling this formats require PHP8+ and PHPxx is used.

Reply:
Note that the check should be based on constants being defined, not on the PHP versions, to avoid issues with the `symfony/polyfill-intl-icu` package (which does not define those constants in all its versions, and I don't even know whether the latest version defines them)

Reply:
They have been added by https://github.com/symfony/polyfill/pull/343, available since [v1.24.0](https://github.com/symfony/polyfill-intl-icu/blob/v1.24.0/IntlDateFormatter.php#L71-L74).